From a2c2409ef32f96fbcccef2e3125581de0eac4381 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Tue, 16 Feb 2021 18:14:57 +0000
Subject: [PATCH] GCC: remove constexpr from operator of FrameToken

MultiToken::value() is not constexpr, but is called from
constexpr operator in FrameToken. However, MultiToken::value()
can't be constexpr, because it returns reference to member.
Remove constexpr from base::UnguessableToken as well to avoid issues
with clang.

Bug: 819294
Change-Id: If75ca8705e9fc46523d61ba3beaac8c96a961a7f
---

diff --git a/base/util/type_safety/token_type.h b/base/util/type_safety/token_type.h
index c9159f28..a101c91 100644
--- a/base/util/type_safety/token_type.h
+++ b/base/util/type_safety/token_type.h
@@ -65,7 +65,7 @@
   template <bool kAllowImplicitConversion2 = kAllowImplicitConversion,
             typename = std::enable_if_t<kAllowImplicitConversion2>>
   // NOLINTNEXTLINE(google-explicit-constructor)
-  constexpr operator const base::UnguessableToken&() const& {
+  operator const base::UnguessableToken&() const& {
     return this->value();
   }
 };
diff --git a/third_party/blink/public/common/tokens/tokens.h b/third_party/blink/public/common/tokens/tokens.h
index 37884d5..a696057 100644
--- a/third_party/blink/public/common/tokens/tokens.h
+++ b/third_party/blink/public/common/tokens/tokens.h
@@ -48,9 +48,7 @@
 
   // Allow implicit casting down to base::UnguessableToken during the migration.
   // NOLINTNEXTLINE(google-explicit-constructor)
-  constexpr operator const base::UnguessableToken&() const& {
-    return this->value();
-  }
+  operator const base::UnguessableToken&() const& { return this->value(); }
 };
 
 ////////////////////////////////////////////////////////////////////////////////
